smdebug/tensorflow/keras.py

```diff
@@ -489,11 +531,8 @@ def _save_metrics(self, batch, logs, force_save=False):
                     self._add_metric(metric_name=key)
                     self._save_for_tensor(key, logs[key], check_before_write=False)
 
-    def _save_layer_input_and_outputs(self, grad_tape=False):
-        # Iterates over all the saved layers for input and output values
-        if is_tf_version_2x() is False or (grad_tape is False and self.model.run_eagerly is False):
-            # This function only works when the run_eagerly is True
-            return
+    def _save_layer_input_and_outputs(self):
+        # Run only for GradTape
```

Comment:
if its only for gradtape, should you check if self.tape is True?

Reply:
Sure